### ISSUE TITLE
FreeBSD support

### ISSUE THREAD

**Issue by jacques-ivy** [wailsapp/wails#2798]
### Is your feature request related to a problem? Please describe.

You currently cannot install the wails cli on FreeBSD via `go install ...` (and even if you could I'm not sure a wails application would work as intended on FreeBSD considering it is not a official target yet)

### Describe the solution you'd like

Official support for FreeBSD(of course also supporting other BSDs would be lovely, but this feature request is about FreeBSD specifically) both to develop on, aswell as for it to be a official application target

**Comment by leaanthony** [comment]
I'm not going to rule it out but will say it's highly unlikely for the following reasons:

- The [number of people who would use it](https://gs.statcounter.com/os-market-share/desktop/worldwide/) would be extremely minimal, perhaps only you!
- None of us have experience with FreeBSD. Is libwebkit installed by default? How do you package it? 
- If we somehow implement FreeBSD then other BSDs may want it too
- Linux is *hard* because it's lagging behind technology wise and it's fragmented as hell

I'm not seeing a compelling argument other than it would be cool, and I'm sure as hell down with that, but it's probably impractical. If you can make a compelling argument for it and can show how it might be practically implemented then please do! 👍

**Comment by trueFireblade** [comment]
Ah I understand and to address some of your points:
- On freebsd we currently have official packages for qt5-webkit, webkit2-gtk3, webkit2-gtk4, aswell as kf5-kdewebkit. I'm not sure which of these (if any) would be helpful to you, though I hope whichever one you use is in that list. Though FreeBSD does not pre-install any graphical things for you, so none of those will be preinstalled on the users system unless an application they use depends on it or they installed it themselves already.
- I get the argument that other BSDs also may want it, but - at least to my knowledge - FreeBSD is the biggest BSD for more desktop-y use-cases (at least of those that still label themselves as BSD and not like e.g. Darwin which are quite decoupled by now) and also you'd get a similar transitivity for BSDs derived from it, as you get over on linux when you create software for Debian/Ubuntu
- Whilst the BSD side also has fragmentation it is more and less at the same time. You have less different "distributions" but those different systems also have less shared components then on the linux side. (you of course also have some that are more so just things like 'more beginner friendly FreeBSD' but way less then linux)
- As for an argument all I can really give you is that it would be quite nice to both be able to use wails myself while I'm not booted into linux, as well as that there is software written in wails I would like to use over here. Though of course I am fully aware that that probably doesn't justify me 'demanding' any work, after all you're doing this on your own time ^^

All in all personally I'd probably recommend you assign this some sort of "low prio" tag and just leave it open to give people the option to have something to reference if they want to implement this (as they also may be using freebsd).

**Comment by istiak101** [comment]
I tried on QEMU and reached this far.

![Screenshot from 2024-10-26 19-35-27](https://github.com/user-attachments/assets/bfb5291a-fd4d-4d14-a7f4-80e1532e7362)

**Comment by leaanthony** [comment]
Incredible! Perhaps a guide on how to do it on freebsd if you're happy to write one? It's unlikely we'd ever support it for practical reasons (no github runners?).

**Comment by COD3HUNT3R** [comment]
@leaanthony I know you have zero plans for FreeBSD. But would you be open to merging patch for v2+v3 as is and call that unsupported and unmaintained?

This is OS specific and should not create any issue for other platforms. So at least Wails can be installed with go install command. v2 support for native items are not that great but v3 support is better. I can send PR against master and v3-alpha branch.

Also if you can tag a release after merging that would be great.

**Comment by leaanthony** [comment]
I'm open to the PR yes - but alpha isn't ready for tagging right now. v2 might be - can check tomorrow.

**Comment by COD3HUNT3R** [comment]
Submitted PR for v2 in master (https://github.com/wailsapp/wails/pull/4341).

v3-alpha waiting on https://github.com/atterpac/refresh/pull/17

**Comment by nbolton** [comment]
> Submitted PR for v2 in master ([#4341](https://github.com/wailsapp/wails/pull/4341)).

@COD3HUNT3R Your PR appears to have been closed last week. Was this intentional?

**Comment by COD3HUNT3R** [comment]
> > Submitted PR for v2 in master ([#4341](https://github.com/wailsapp/wails/pull/4341)).
> 
> [@COD3HUNT3R](https://github.com/COD3HUNT3R) Your PR appears to have been closed last week. Was this intentional?

Maintainer likes to wastes people time when clearly they do not intend to support it officially.

https://github.com/Sammy-T/gotepad/issues/118#issuecomment-3091951345

**Comment by leaanthony** [comment]
There are other PRs in the pipeline before yours. Feel free to reopen it if you want it to be considered.

**Comment by nbolton** [comment]
> Maintainer likes to wastes people time

@COD3HUNT3R Please remember that the maintainer is doing this work for the good of the community. I strongly doubt they have any intention to waste anyone's time. From what I can see from the comment you linked to, the maintainer appears supportive. Please reconsider reopening your PR and sticking with it. I know there are a lot of code quality comments on the PR, and that can be frustrating, but this will help to maintain high quality software.

@leaanthony Thank you for your hard work and dedication.

**Comment by leaanthony** [comment]
Thanks @nbolton! Also thanks for Synergy!

**Comment by COD3HUNT3R** [comment]
> > Maintainer likes to wastes people time
> 
> [@COD3HUNT3R](https://github.com/COD3HUNT3R) Please remember that the maintainer is doing this work for the good of the community. I strongly doubt they have any intention to waste anyone's time. From what I can see from the comment you linked to, the maintainer appears supportive. Please reconsider reopening your PR and sticking with it. I know there are a lot of code quality comments on the PR, and that can be frustrating, but this will help to maintain high quality software.

Although it seems this is my first PR it is not. They might have good faith but I do not want continue working on this anymore. I do not work on FreeBSD exclusively and I already have internal forks so this is also my effort for the FreeBSD community. Feel free to take my work and polish it.

Note: About code quality this is same implementation as Linux except few small changes which already exists on the Wails codebase. Should I refactor those too?

**Comment by nbolton** [comment]
> Feel free to take my work and polish it.

Good idea. I was contemplating this. FreeBSD is a great OS (I'm still learning the ropes, though).

> Although it seems this is my first PR it is not.

Oh that was actually not my impression, sorry if my comment came across this way.

> About code quality this is same implementation as Linux except few small changes which already exists on the Wails codebase. Should I refactor those too?

@leaanthony What's the general guidance when it comes to addressing all these AI comments and Sonar Scanner? Should they be ignored as they're relating to existing code, fixed in another PR, or something else?

**Comment by leaanthony** [comment]
The scanner I'm less worried about as it generally moans about duplicate code in templates 😩 It's a big PR so likely the rabbit will find things. If you don't think it's a problem just comment and close. Often it finds very useful things

**Comment by leaanthony** [comment]
Also want to reiterate that this would be community maintained unless by some miracle I win the lottery and can do this full time.

**Comment by leaanthony** [comment]
This feature request has been added to our v3 Backlog. As v3 is now our primary focus, all feature development is being directed there.

You can track progress on the v3 Backlog project board: https://github.com/orgs/wailsapp/projects/13

Thanks for your patience\!

**Comment by leaanthony** [comment]
Thanks for the request. FreeBSD/BSD isn't an officially supported target for Wails v3, and there are no current plans to add it. While the CLI itself is portable Go, a working app needs platform backends for the window, webview (WebKitGTK), menus, tray, dialogs, clipboard, screens, etc. — all currently Linux/macOS/Windows-only. That's a substantial ongoing maintenance commitment we can't take on right now.

Closing as not-planned — but a well-scoped community PR adding a FreeBSD backend would be very welcome, and we're happy to give pointers if someone wants to attempt it. Thanks!

**Comment by nbolton** [comment]
Yeah that's fair, I can see the struggle. You'd need a dedicated BSD maintainer on the product and those are rarer than hen's teeth!
